### PR TITLE
Remove the outdated rounding code

### PR DESCRIPTION
**Pull request by johnmccombs1**
# Description
This code is causing additional rounding problems for a merchant 

Fixes: 
https://app.asana.com/0/1125081140214268/1168686308499785
https://app.asana.com/0/1125081140214268/1164552839717137

#changelog Remove the outdated rounding code

# Type of change

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [x] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: Helper/Cart.php [hunk 1]
```diff
@@ -1386,9 +1386,6 @@ public function getCartData($paymentOnly, $placeOrderPayload, $immutableQuote =
             $taxAmount = 0;
         }
 
-        // include potential rounding difference and reset $diff accumulator
-        $cart['items'][0]['total_amount'] += round($diff);
-        $totalAmount += round($diff);
         $diff = 0;
```
Comment (danac-gs) [inline Helper/Cart.php]: This line should go as well. We apply this rounding diff below in this method. Applying it twice seems redundant and could probably cause errors. So all 3 lines above should be removed.

Comment (johnmccombs1) [reply Helper/Cart.php]: @dan-bolt I've updated the code

**Comment by danac-gs** [comment]
@johnmccombs1 @matt-thomason @njfaries  @daisy1754 @OleksiiOnyshchenko 
The rounding issue that merchant had here is a very rare one and hard to replicate and foresee. I believe this is a good fix, applied to merchant store already, and John and I were discussing it. He was cautious about the regression and my standpoint was to go on with it let QA do extensive testing. So, I suggest paying special attention when doing acceptance testing and spending some time on it with diverse carts. For example, cart items should have various decimal number prices, discounts should be applied, especially percentage type discounts, different ones, many combinations of these should be taken into account. One should just spend a couple of hours making random orders with complex carts. And if the case that does not work is found we should have the exact cart specification for debugging purposes.

**Comment by daisy1754** [comment]
okay your call - still don't understand why test can be added though

**Comment by danac-gs** [comment]
> okay your call - still don't understand why test can be added though

We should add unit test, of course, but can not be sure to catch this case in them. It is a functional / integration issue that pops up later, in order creation.

**Pull request merged**